specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerActiveConfiguration.json

```diff
@@ -0,0 +1,171 @@
+          "required": false,
+          "type": "string",
+          "x-ms-parameter-location": "method"
+        }
+      ],
+      "get": {
+        "tags": [
+          "NetworkManagerActiveConfigurations"
+        ],
+        "operationId": "ActiveConfigurations_List",
+        "description": "Lists active configurations in a network manager",
+        "responses": {
+          "200": {
+            "description": "Get successful. The operation returns the specified Managed Network active configuration resource.",
+            "schema": {
+              "$ref": "#/definitions/ActiveConfigurationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Active Configurations": {
+            "$ref": "./examples/NetworkManagerActiveConfigurationList.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ActiveConfiguration": {
+      "properties": {
+        "commitTime": {
+          "type": "string",
+          "description": "Deployment time string",
+          "format": "date-time"
+        },
+        "region": {
+          "type": "string",
+          "description": "deployment region."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./networkManagerEffectiveConfiguration.json#/definitions/EffectiveConfiguration"
+        }
+      ],
+      "description": "Active Configuration."
+    },
+    "ActiveConfigurationListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ActiveConfiguration"
+          },
+          "description": "Gets a page of ActiveConfigurations"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next set of results."
+        }
+      },
+      "description": "Result of the request to list activeConfigurations. It contains a list of groups and a URL link to get the next set of results."
+    }
+  },
+  "parameters": {
+    "ResourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the resource group.",
+      "x-ms-parameter-location": "method"
+    },
+    "NetworkManagerNameParameter": {
+      "name": "networkManagerName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the network manager.",
+      "x-ms-parameter-location": "method"
+    },
+    "ListPageSizeParameter": {
+      "name": "$top",
+      "description": "May be used to limit the number of results in a page for list queries.",
```

Comment:
This parameter (and the description) seems ambiguous:

- The name `ListPageSizeParameter` and the description conveys that it could be used to limit the size of each page when paging through results using `nextLink`/`skipToken`. 
- But the name `$top`  usually means "query to only return the top N results based by some ordering criteria", which is independent of page size. 

I am willing to live with the ambiguity if this is a well-known idiom in Azure, but I did a quick search for other APIs and I couldn't find evidence. 

Assuming we follow the traditional semantics of top, then I suggest we rename `ListPageSizeParameter` and update the description to something more straightforward like:

```suggestion
      "description": "May be used to limit the results to the top N active configurations.",
```

I borrowed this from other Azure APIs that accept `$top`. 

However even this leaves something to be desired because the user cannot know which N active configurations will be returned. 

Is it the most recently created, the least recently created, or the top N by some other sort criteria? If it was for example the most recently created active configurations, I would suggest the description to be:

```suggestion
      "description": "May be used to limit the results to the N most recent active configurations.",
```

If the semantics we want is that this parameter is to limit the page size, then I would suggest we rename `$top` to something different like `$pageSize`  or `$take`.

We should at least pick one or the other, but I believe having both parameters separately would be better because they are useful for different reasons.

**Note that this comment applies to other APIs that accept the "$top"  argument.**

Reply:
The name $top is required by https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#get-resource. I will update comments. Thanks!

Reply:
Ok, if we are following normal (e.g. OData) `$top` semantics you not just need to update the comments. You should also rename `ListPageSizeParameter` to something that conveys the semantics, e.g. `TopParameter`.